### PR TITLE
Group Quota SDK 

### PR DESCRIPTION
**Pull request by tejasm-microsoft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by jsquire — CHANGES_REQUESTED** [review]
@tejasm-microsoft :

Your account lacks the required public GitHub organizations and permissions required of an internal contributor.  Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

File: sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs [hunk 1]
```diff
@@ -532,8 +532,8 @@ private static bool IsFinalState(Response response, HeaderSource headerSource, o
                         switch (headerSource)
                         {
                             case HeaderSource.None when root.TryGetProperty("properties", out var properties) && properties.TryGetProperty("provisioningState", out JsonElement property):
-                            case HeaderSource.OperationLocation when root.TryGetProperty("status", out property):
-                            case HeaderSource.AzureAsyncOperation when root.TryGetProperty("status", out property):
+                            case HeaderSource.OperationLocation when root.TryGetProperty("properties", out properties) && properties.TryGetProperty("provisioningState", out property):
```
Comment (jsquire) [inline sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs:535]: This needs to be reverted.   You should not be making changes to Azure Core files here.

**Review by jsquire — DISMISSED** [review]

File: sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs (hunk 1 above)
Comment (tejasm-microsoft) [reply sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs]: Reverted the changes made in the core files

File: NuGet.Config [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <!-- Do not add any additional feeds if new packages are needed they need to come from nuget.org or our azure-sdk-for-net DevOps feed -->
     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
     <add key="azure-sdk-for-net" value="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json" />
+    <add key="Group Quota Nuget" value="C:\GroupQuotaNuget" />
```
[the remark this answers is not in the document]
Comment (tejasm-microsoft) [reply NuGet.Config]: Removed this extra package

**Comment by tejasm-microsoft** [comment]
> @tejasm-microsoft :
> 
> Your account lacks the required public GitHub organizations and permissions required of an internal contributor. Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
> 
> * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> 
> You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.
> 
> Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

Added myself to the code owners file, not sure if I followed the right format

Also there seems to be a discrepancy between the output of the Validate-AzsdkCodeOwner, since it seems I am apart of the two required orgs on the github portal?
image

**Comment by jsquire** [comment]
> Also there seems to be a discrepancy between the output of the Validate-AzsdkCodeOwner, since it seems I am apart of the two required orgs on the github portal?

@tejasm-microsoft :  Your organizations are not marked as public.   You see them, as to other members of those orgs but they are private on your account.   Please follow the onboarding guidance.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -968,6 +968,9 @@
 # PRLabel: %Spring App Discovery
 /sdk/springappdiscovery/Azure.ResourceManager.*/                   @sunkun99 @archerzz @ArcturusZhang @ArthurMa1978
 
+#PRLabel: %Quota
+/sdk/quota/Azure.ResourceManager.*/                                @tejasm-microsoft
```
Comment (jsquire) [inline .github/CODEOWNERS:972]:
```suggestion
# PRLabel: %Quota
/sdk/quota/Azure.ResourceManager.*/                                @tejasm-microsoft

# ServiceLabel: %Quota %Mgmt
# ServiceOwners:                                                   @tejasm-microsoft
```

These need to follow the same format as the other management libraries.   Please move this block - with my suggestion - up to ~L959 to preserve alphabetical ordering by label.

Comment (tejasm-microsoft) [reply .github/CODEOWNERS]: Followed the order suggested and was able to verify via PS command:
image

**Comment by tejasm-microsoft** [comment]
Added myself to CODEOWNERS file and verified sufficient permissions via Powershell commands:
image
@jsquire

**Comment by live1206** [comment]
- The tests are failing [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3963424&view=ms.vss-test-web.build-test-results-tab&runId=48830982&resultId=103974&paneView=debug), please use test-proxy to upload the recordings, [details](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md#test-proxy).
> Generated code is not up to date.
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code and samples.
- CI is failing as above, please make sure you merge the latest main and regen SDK and export the API

**Comment by tejasm-microsoft** [comment]
> * The tests are failing [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3963424&view=ms.vss-test-web.build-test-results-tab&runId=48830982&resultId=103974&paneView=debug), please use test-proxy to upload the recordings, [details](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md#test-proxy).
> 
> > Generated code is not up to date.
> > You may need to rebase on the latest main,
> > run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets),
> > run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions).
> > run 'dotnet build /t:GenerateCode' to update the generated code and samples.
> 
> * CI is failing as above, please make sure you merge the latest main and regen SDK and export the API

Resolved the failing tests  and added the assets.json via test proxy:
https://github.com/Azure/azure-sdk-assets/tree/net/quota/Azure.ResourceManager.Quota_09eadc6fbb/net/sdk/quota/Azure.ResourceManager.Quota/tests/SessionRecords/GroupQuotaTests

Also ran Code Checks script locally and came back with no errors

**Comment by live1206** [comment]
> > > Generated code is not up to date.
> > > You may need to rebase on the latest main,
> > > run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets),
> > > run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions).
> > > run 'dotnet build /t:GenerateCode' to update the generated code and samples.
> > 
> > 
> > 
> > * CI is failing as above, please make sure you merge the latest main and regen SDK and export the API
> 
> Resolved the failing tests and added the assets.json via test proxy: https://github.com/Azure/azure-sdk-assets/tree/net/quota/Azure.ResourceManager.Quota_09eadc6fbb/net/sdk/quota/Azure.ResourceManager.Quota/tests/SessionRecords/GroupQuotaTests
> 
> Also ran Code Checks script locally and came back with no errors

I merged the latest main to your branch and re-generate the SDK for you, the corresponding CI should be fixed now.

The tests are still failing, you can reproduce the failing tests by running the tests locally.
- Remove the local recording located in `\src\azure-sdk-for-net\.assets`
- Open a cmd under `\src\azure-sdk-for-net\sdk\quota\Azure.ResourceManager.Quota`, restore the remote recordings by running "test-proxy restore -a ./assets.json"
- run the tests, you should see the test failure

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**